sympy/logic/boolalg.py

```diff
@@ -1344,7 +1342,7 @@ def __new__(cls, *args, **options):
         for x in args:
             if isinstance(x, Number) or x in [True, False]:  # Includes 0, 1
                 argset.discard(x)
-                argset.add(True if x else False)
+                argset.add(bool(x))
```

Comment:
The condition should probably be `if x in (S.Zero, S.One)`.

I think it's a mistake to accept numbers here altogether.